packs/mmonit/sensors/events_sensor.py

```diff
@@ -0,0 +1,57 @@
+import requests
+from st2reactor.sensor.base import PollingSensor
+
+
+class MmonitEventsSensor(PollingSensor):
+    def __init__(self, sensor_service, config=None, poll_interval=5):
+        super(MmonitEventsSensor, self).__init__(sensor_service=sensor_service, config=config, poll_interval=poll_interval)
+
+    def setup(self):
+        self.url = self._config["shared_sensors_config"]["host"].strip("/")
+        self.user = self._config["shared_sensors_config"]["user"]
+        self.password = self._config["shared_sensors_config"]["password"]
+        self.session = requests.session()
+        self._login()
+
+    def _login(self):
+        self.session.get(self.url)
+        login = self.session.get("{}/z_security_check".format(self.url), data={"z_csrf_protection": "off",
+                                                                               "z_username": self.user,
+                                                                               "z_password": self.password})
+        if login.status_code != 200:
+            raise Exception("Could not login to mmonit {}".format(login.reason))
+
+    def poll(self):
+        # always check for status_code on a secured page on each poll in case we get logged out
+        test_login = self.session.get("{}/session/get".format(self.url))
+        if test_login.status_code != 200:
+            self._login()
+
+        events_list = self.session.get("{}/reports/events/list").json()
+        events = self._clear_list(events_list)
+        for event in events:
+            payload = {"host": event['hostname'], "id": event['id'],
+                       "event": event['event'], "servicename": event["servicename"],
+                       "full_data": event}
+            self._sensor_service.dispatch(trigger="mmonit.new_event", payload=payload)
+            self._sensor_service.set_value('monit.last_event', event["id"])
```

Comment:
That has the side effect of triggering duplicated events if something goes wrong. So you have 50 events, you are going trougth the number 28, somebody restarts the sensor service or it dies for some reason, then your last_event stored was not the 28 but something complete different and the sensor will trigger those events again.